### PR TITLE
applications: serial_lte_modem: Coverity Fix out-of-bounds write

### PR DESCRIPTION
**Pull request by junqingzou**
Add util function util_string_get() to call at_params_string_get()
and null-treminate the result string.

Plus code cleaning of unnecessary check of AT param count.

Signed-off-by: Jun Qing Zou <jun.qing.zou@nordicsemi.no>

### CODE REVIEW

File: applications/serial_lte_modem/src/slm_util.c [hunk 1]
```diff
@@ -173,3 +172,25 @@ bool util_get_ipv4_addr(char *address)
...
+	int ret;
+	size_t size = *len;
+
+	/* at_params_string_get calls "memcpy" instead of "strcpy" */
+	ret = at_params_string_get(list, index, value, len);
+	if (ret) {
+		return ret;
+	}
+	if (*len < size) {
+		*(value + *len) = '\0';
```
Comment (rlubos) [inline applications/serial_lte_modem/src/slm_util.c:191]: Shouldn't `len` be incremented since NULL termination is added?

Comment (junqingzou) [reply applications/serial_lte_modem/src/slm_util.c:191]: We're simulating strcpy(), and "len" is not used outside this function.

File: applications/serial_lte_modem/src/slm_util.h [hunk 2]
```diff
@@ -101,6 +102,24 @@ bool check_for_ipv4(const char *address, uint8_t length);
...
 
+/**
+ * @brief Get string value from AT command with length check.
+ *
+ * @p len must be bigger than the string length, or an error is returned.
+ * The copied string is null-terminated.
+ *
+ * @param[in]     list    Parameter list.
+ * @param[in]     index   Parameter index in the list.
+ * @param[in]     value   Pointer to the buffer where to copy the value.
```
[the remark this answers is not in the document]
Comment (junqingzou) [reply applications/serial_lte_modem/src/slm_util.h:113]: Yes, copied from at_params_string_get() :)

**Comment by junqingzou** [comment]
@rlubos I'll break the PR into two commits, one to remove redundant count check, and second the Coverity fix

File: applications/serial_lte_modem/src/slm_util.c (hunk 1 above)
Comment (junqingzou) [reply applications/serial_lte_modem/src/slm_util.c:191]: We're simulating strcpy() just to help strlen() after the function call. Also outside the function, the "len" is only directly used after parsing the data to send uplink, in which context we should not include the null-terminator and thus not incremented

**Review by lats1980 — APPROVED** [review]
Looks good to me.

File: applications/serial_lte_modem/src/slm_util.c (hunk 1 above)
Comment (junqingzou) [reply applications/serial_lte_modem/src/slm_util.c:191]: You're right, this is misleading. Updated

File: applications/serial_lte_modem/src/slm_util.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (junqingzou) [reply applications/serial_lte_modem/src/slm_util.h:114]: Understood, updated accordingly

**Review by rlubos — APPROVED** [review]

**Pull request merged**